src/kafka/kafka_hello_world.clj

```diff
@@ -0,0 +1,98 @@
+                           "group.id",          "group1"
+                           "bootstrap.servers", "localhost:9092"
+                           "key.deserializer"   StringDeserializer
+                           "value.deserializer" StringDeserializer}}))
+
+
+
+(defn build-producer
+  []
+  (KafkaProducer. (get @properties :producer)))
+
+(defn build-consumer
+  [topic]
+  (let [consumer         (KafkaConsumer. (get @properties :consumer))
+        _subscribe-topic (.subscribe consumer [topic])]
+    consumer))
+
+(def producer-consumer
+  (let [producer          (build-producer)
+        consumer          (build-consumer "example-topic")
+        producer-consumer (atom [producer consumer])]
+    producer-consumer))
+
+(defn send-msg!
+  [producer msg-value]
+  (let [producer-configuration "example-topic"
+        msg                    (ProducerRecord. producer-configuration "msg" msg-value)
+        send-msg               (.send producer msg)]
+    send-msg))
+
+(defn receive-messages
+  [consumer]
+  (let [records  (.poll consumer (Duration/ofMillis 100000))
+        messages (->> records
+                      (mapv (fn [record]
+                              (str (.value record)))))]
+    messages))
+
+(comment
+
+  ;;------------------------------function evaluation----------------------------------
+  (deref properties)
+  #_=> {:producer-config {"client.id"         "m-p7p1q34c6p",
+                          "group.id"          "group1",
+                          "bootstrap.servers" "localhost:9092",
+                          "key.serializer"    org.apache.kafka.common.serialization.StringSerializer,
+                          "value.serializer"  org.apache.kafka.common.serialization.StringSerializer},
+        :consumer-config {"client.id"          "m-p7p1q34c6p",
+                          "group.id"           "group1",
+                          "bootstrap.servers"  "localhost:9092",
+                          "key.deserializer"   org.apache.kafka.common.serialization.StringDeserializer,
+                          "value.deserializer" org.apache.kafka.common.serialization.StringDeserializer}}
+
+
+  (build-producer)
+  #_=> #_#object[org.apache.kafka.clients.producer.KafkaProducer
+                 0xd7d9d44
+                 "org.apache.kafka.clients.producer.KafkaProducer@d7d9d44"]
+
+  (build-consumer "example-topic")
+  #_=> #_#object[org.apache.kafka.clients.consumer.KafkaConsumer
+                 0x5e1f9219
+                 "org.apache.kafka.clients.consumer.KafkaConsumer@5e1f9219"]
+
+  (deref producer-consumer)
+  #_=> #_[#object[org.apache.kafka.clients.producer.KafkaProducer
+                  0x301c93e1
+                  "org.apache.kafka.clients.producer.KafkaProducer@301c93e1"]
+          #object[org.apache.kafka.clients.consumer.KafkaConsumer
+                  0x4f2f21e6
+                  "org.apache.kafka.clients.consumer.KafkaConsumer@4f2f21e6"]]
+
+
+  (send-msg! (build-producer) "Message content")
```

Comment:
Building a producer every time you send a message? See above for saving state in atoms